### PR TITLE
A fresh start on dependencies

### PR DESCRIPTION
**Pull request by rowanseymour**
So yes... there's a reasonably high chance this will break something but it does seem like it's worth the pain to do this every few years. We haven't kept pip-requires.txt and pip-freeze.txt in sync and there's stuff in both files that's no longer used. 

New pip-requires.txt is a bit more organized and new pip-freeze.txt is generated by [pip-tools](https://github.com/jazzband/pip-tools) so it shows who is depending on who.

Anything with a pinned version in pip-requires.txt is something that caused problems with the latest version, and we should look at each of those at some point to see what it takes to update them.

(this PR replaces https://github.com/rapidpro/rapidpro/pull/530)

### CODE REVIEW

File: temba/orgs/tests.py [hunk 1]
```diff
@@ -2369,7 +2369,7 @@ def test_language_codes(self):
         self.assertEquals('French, Old (842-ca.1400)', matches[3]['text'])
 
         # try a language that doesn't exist
-        self.assertEquals(None, languages.get_language_name('klingon'))
+        self.assertEquals(None, languages.get_language_name('xyz'))
```
Comment (rowanseymour) [inline temba/orgs/tests.py]: Turns out Klingon is a language now

File: setup.cfg [hunk 2]
```diff
@@ -2,4 +2,4 @@
 max-line-length=120
 filename=*.py,*.py.dev
 exclude=./env/*,./config/*,./fab*,.git/*,static/bower/*
-ignore=E501,T003
+ignore=E501,F405,T003
```
Comment (rowanseymour) [inline setup.cfg]: This is flake8's new stricter checking of star imports that we still need in our settings files

File: temba/urls.py [hunk 3]
```diff
@@ -63,6 +63,7 @@ def init_analytics():
     if librato_user and librato_token:  # pragma: needs cover
         init_librato(librato_user, librato_token)
 
+
```
Comment (rowanseymour) [inline temba/urls.py]: flake8 now demands two spaces after a function definition

File: temba/orgs/tests.py (hunk 1 above)
Comment (norkans7) [inline temba/orgs/tests.py]: 🤣

File: temba/urls.py (hunk 3 above)
Comment (nicpottier) [inline temba/urls.py]: So useful!

**Comment by nicpottier** [comment]
Let's do it!

**Comment by rowanseymour** [comment]
I'll push this to staging on Monday and test everything I can think of

**Comment by smn** [comment]
Awesome, thanks @rowanseymour!

File: pip-requires.txt [hunk 4]
```diff
@@ -52,7 +52,6 @@ gunicorn
 iptools
 iso639
 isoweek
-numpy
```
Comment (smn) [inline pip-requires.txt]: ❤️

**Pull request merged**